### PR TITLE
Small bug fix and enhancement

### LINKED ISSUE #41: RNG seed hard-coded
integerise function sets RNG seed to hard-coded value every time its called, so always gives same result.
suggest the seed is passed as an optional argument, with default value same as current

### PR DESCRIPTION
**Pull request by virgesmith**
Fixes #41 and #42

### CODE REVIEW

**Comment by philmikejones** [comment]
Thanks for the pull request @virgesmith I've seen the Travis build has failed so will look into this tonight. Looks like a unit test has failed.

**Comment by virgesmith** [comment]
No probs - trying to work out why travis failed too. Checks and tests were all fine for me. Relevant bit of travis log seems to be
`* checking examples ... ERROR
Running examples in ‘rakeR-Ex.R’ failed
The error most likely occurred in:

> base::assign(".ptime", proc.time(), pos = "CheckExEnv")
> ### Name: integerise
> ### Title: integerise
> ### Aliases: integerise
> 
> ### ** Examples
> 
> cons <- data.frame(
+   "zone"   = letters[1:3],
+   "a0_49"  = c(8, 2, 7),
+   "a_gt50" = c(4, 8, 4),
+   "f"      = c(6, 6, 8),
+   "m"      = c(6, 4, 3)
+ )
> 
> inds <- data.frame(
+   "id"     = LETTERS[1:5],
+   "age"    = c("a_gt50", "a_gt50", "a0_49", "a_gt50", "a0_49"),
+   "sex"    = c("m", "m", "m", "f", "f"),
+   "income" = c(2868, 2474, 2231, 3152, 2473),
+   stringsAsFactors = FALSE
+ )
> vars <- c("age", "sex")
> 
> weights     <- weight(cons = cons, inds = inds, vars = vars)
Error: all.equal(colnames(cons), colnames(ind_cat)) is not TRUE
Execution halted'

which I can't reproduce...

**Comment by philmikejones** [comment]
The latest build failed because of [this issue](https://github.com/travis-ci/travis-ci/issues/7332). Restarted.

**Comment by philmikejones** [comment]
There's a problem with the documentation example for the `integerise()` function when building on Travis, but I can't reproduce it locally. I need to take a look at this later when I'm less tired!

File: R/rake_functions.R [hunk 1]
```diff
@@ -287,10 +287,10 @@ extract_weights <- function(weights, inds, id) {
 #' weights     <- weight(cons = cons, inds = inds, vars = vars)
 #' weights_int <- integerise(weights)
 #' weights_int
-integerise <- function(weights, method = "trs") {
+integerise <- function(weights, method = "trs", seed = 42) {
```
Comment (philmikejones) [inline R/rake_functions.R:290]: Do we even need to put a manual seed in here? Is the result of `set.seed()` not just doing the same thing, now I think about it?

File: R/rake_functions.R [hunk 2]
```diff
@@ -311,11 +311,13 @@ integerise <- function(weights, method = "trs") {
   weights_dec <- weights_vec - weights_int
   deficit <- round(sum(weights_dec))
 
-  # the weights be 'topped up' (+ 1 applied)
-  topup <- sample(length(weights), size = deficit, prob = weights_dec)
-
-  weights_int[topup] <- weights_int[topup] + 1
+  # do nothing if weights are already integers (sample will throw)
```
Comment (philmikejones) [inline R/rake_functions.R:314]: Sorry, not sure what you mean by 'sample will throw'? @virgesmith

**Comment by philmikejones** [comment]
Builds on Precise are working (although now there's a new problem). I'm going to leave this a couple of days until Travis settles down then come back to it. Thank you for your suggestions @virgesmith , I will get around to them when systems are working again!

File: R/rake_functions.R (hunk 1 above)
Comment (philmikejones) [reply R/rake_functions.R:290]: Calling seed as an argument with a default fixes the seed (which can be manually changed).
Without it the seed changes unless it's manually set.

**Review by philmikejones — APPROVED** [review]

**Pull request merged**